packages/ecc-client-ga4gh-tes/src/components/tes-create-run/definition/tesCreateRun.ts

```diff
@@ -0,0 +1,579 @@
+   */
+  handleEnvNameChange = (
+    event: Event,
+    executor: ExecutorData,
+    index: number
+  ) => {
+    const newEnvName = (event.target as HTMLInputElement).value;
+    const entries = Object.entries(executor.env);
+    entries[index][0] = newEnvName;
+    executor.env = Object.fromEntries(entries);
+  };
+
+  /**
+   * Handles the change of the value field of env of the executor field
+   * @param event The event triggered when the value field of env from any executor changes
+   * @param executor The specific executor in the taskExecutors that is being changed
+   * @param index The index of the env being changes if considered as an array
+   */
+  handleEnvValueChange = (
+    event: Event,
+    executor: ExecutorData,
+    index: number
+  ) => {
+    const newEnvValue = (event.target as HTMLInputElement).value;
+    const entries = Object.entries(executor.env);
+    entries[index][1] = newEnvValue;
+    executor.env = Object.fromEntries(entries);
+  };
+
+  /**
+   * Handles the stderr input for the executor
+   * @param event The event triggered when the stderr field of the executor field changes
+   * @param executor The specific executor in the taskExecutors that is being changed
+   */
+  handleExecutorsStderrChange = (event: Event, executor: ExecutorData) => {
+    const stderrInput = (event.target as HTMLInputElement).value;
+    executor.stderr = stderrInput;
+  };
+
+  /**
+   * Handles the stdout input for the executor
+   * @param event The event triggered when the stdout field of the executor field changes
+   * @param executor The specific executor in the taskExecutors that is being changed
+   */
+  handleExecutorsStdoutChange = (event: Event, executor: ExecutorData) => {
+    const stdoutInput = (event.target as HTMLInputElement).value;
+    executor.stdout = stdoutInput;
+  };
+
+  /**
+   * Handles the stdin input for the executor
+   * @param event The event triggered when the stdin field of the executor field changes
+   * @param executor The specific executor in the taskExecutors that is being changed
+   */
+  handleExecutorsStdinChange = (event: Event, executor: ExecutorData) => {
+    const stdinInput = (event.target as HTMLInputElement).value;
+    executor.stdin = stdinInput;
+  };
+
+  /**
+   * Handles the workdir input for the executor
+   * @param event The event triggered when the workdir field of the executor field changes
+   * @param executor The specific executor in the taskExecutors that is being changed
+   */
+  handleExecutorsWorkdirChange = (event: Event, executor: ExecutorData) => {
+    const workdirInput = (event.target as HTMLInputElement).value;
+    executor.workdir = workdirInput;
+  };
+
+  /**
+   * Populate more executor fields
+   */
+  addExecutor = () => {
+    this.taskExecutors.push({ ...executorTemplate });
+    this.taskExecutorsLength += 1;
+  };
+
+  // Custom comparison function for executors
+  areExecutorsEqual = (executor1: ExecutorData, executor2: ExecutorData) => {
```

Comment:
this seems like an overly complex thing to do when adding a unique id will allow you to compare the executors much more easily. And if you end up changing some of the data in the executors or just how some of the data is structured you'll have to modify this function too.

Reply:
I understand, but the thing is this will require changing the type, and then re-conversion of the data to the data without an id, it was getting really messy because the API straight out rejects to create a response if there are any issues. Look into the last implementation, it did have an id but the implementation got really messy as each kind of data dat multiple data type.

Reply:
okay, but you'll need to make this a helper function in the future and maybe make it more reusable incase it's needed somewhere else too. And maybe look into a package like lodash but only if you have a use for more of it's functionality than just this one function (it also provides a way to completely deep copy an object with no side effects)